lib/rules/no-case-declarations.js

```diff
@@ -47,6 +47,13 @@ module.exports = {
 
         return {
             SwitchCase(node) {
+                if (
+                    node.consequent.length === 0 || // Empty clause
+                    node.consequent[0].type === "BlockStatement" && node.consequent.length === 1 // Clause has {} around child expressions
```

Comment:
Better to check "the only element" part earlier
```suggestion
                    node.consequent.length === 1 && node.consequent[0].type === "BlockStatement"  // Clause has {} around child expressions
```